practice7/MaybeTheBestCodeEver.cpp

```diff
@@ -0,0 +1,156 @@
+
+#include <iostream>
+#include <vector>
+#include <string>
+#include <algorithm>
+#include <cstdio>
+#include <filesystem>
+//#include <print>
+
+int gCount = 0;
+double gBiggestArea = 0;
+double gSmallestArea = 1000000;
+
+class Rectangle {
+public:
+    double Width;
+    double Height;
+    std::string Name;
+
+    Rectangle(const std::string& n, double w, double h) : Name(n), Width(w), Height(h) {}
+};
+
+// Функція для створення вектора об'єктів Rectangle
+std::vector<Rectangle> createRectangles(int count) {
+    std::vector<Rectangle> rectangles;
+
+    for (int i = 0; i < count; ++i) {
+        double width, height;
+        std::string name = "Rectangle " + std::to_string(i + 1);
+        //std::println("Enter width and height for {}  : ", name());
+       	std::cout <<"Enter width and height for " << name << ":" ;
+	std::cin >> width >> height;
+
+        rectangles.emplace_back(name, width, height);
+    }
+
+    return rectangles;
+}
+//функція для пошуку прямокутника з максимальною площею. повертає його назву
+std::string findRectangleWithMaxArea(const std::vector<Rectangle>& rectangles) {
+    if (rectangles.empty()) {
+        return "No rectangles provided";
+    }
+
+    const Rectangle* maxAreaRect = &rectangles[0];
+    double maxArea = rectangles[0].Width * rectangles[0].Height;
```

Comment:
it is reasonable to have area calculation inside rectangle class and hide it for user of the class. some getArea method can be used to get the value